src/python/pants/backend/codegen/protobuf/buf.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import List, cast
+
+from pants.core.util_rules.external_tool import TemplatedExternalTool
+from pants.engine.platform import Platform
+
+
+class Buf(TemplatedExternalTool):
+    options_scope = "buf"
```

Comment:
Because this is such a short name, it raised my guard if it's dangerous for us to squat on the name `buf`. Whether we should use `buf-linter` for example. But I think it's unlikely we'll have another tool/subsystem named `buf`, so this seems fine. And we can deprecate the name if we do have eventual ambiguity. Wdyt?

Reply:
I changed it to `buf-lint` to keep it in line with `skip_buf_lint`.